opnode/rollup/driver/state.go

```diff
@@ -173,19 +175,33 @@ func (s *state) findNextL1Origin(ctx context.Context) (eth.L1BlockRef, error) {
 	}
 	// Somehow reorg'd. Will let the state loop take care of it.
 	if curr.Hash != s.l2Head.L1Origin.Hash {
-		return eth.L1BlockRef{}, errors.New("Unknown L1Origin")
+		return eth.L1BlockRef{}, errors.New("unknown L1Origin")
+	}
+
+	nextOrigin, err := s.l1.L1BlockRefByNumber(ctx, curr.Number+1)
+	if errors.Is(err, ethereum.NotFound) {
+		// no new L1 origin found, keep the current one
+		s.log.Info("No new L1 origin, staying with current one", "l2Head", s.l2Head, "l1Origin", curr)
+		return nextOrigin, nil
```

Comment:
```suggestion
		return curr, nil
```

Maybe also rename `curr` to `currentOrigin`